src/api/models/bedrock.py

```diff
@@ -738,35 +738,53 @@ def _create_response_stream(
 
         return None
 
-    def _parse_image(self, image_url: str) -> tuple[bytes, str]:
-        """Try to get the raw data from an image url.
+    def _get_supported_image_types(self) -> list[str]:
+        return ["png", "jpeg", "gif", "webp"]
 
-        Ref: https://docs.aws.amazon.com/bedrock/latest/APIReference/API_runtime_ImageSource.html
-        returns a tuple of (Image Data, Content Type)
-        """
-        pattern = r"^data:(image/[a-z]*);base64,\s*"
-        content_type = re.search(pattern, image_url)
-        # if already base64 encoded.
-        # Only supports 'image/jpeg', 'image/png', 'image/gif' or 'image/webp'
-        if content_type:
-            image_data = re.sub(pattern, "", image_url)
-            return base64.b64decode(image_data), content_type.group(1)
-
-        # Send a request to the image URL
-        response = requests.get(image_url)
-        # Check if the request was successful
-        if response.status_code == 200:
+    def _parse_image(self, image_url: str) -> tuple[bytes, str]:
+        
+        if "base64" in image_url:
```

Comment:
Could you improve this check for the edge case like `https://example.com/path/base64.jpg`?